tools/bag_processing/scripts/convert_bayer_to_grayscale.py

```diff
@@ -30,6 +30,31 @@
 from cv_bridge import CvBridge, CvBridgeError
 from sensor_msgs.msg import Image
 
+
+def convert_bayer_to_grayscale(
+    bagfile, bayer_image_topic, gray_image_topic, save_all_topics=False
+):
+    bridge = CvBridge()
+    topics = None if save_all_topics else [bayer_image_topic]
+    output_bag_name = os.path.splitext(bagfile)[0] + "_gray.bag"
+    output_bag = rosbag.Bag(output_bag_name, "w")
+
+    with rosbag.Bag(bagfile, "r") as bag:
+        for topic, msg, t in bag.read_messages(topics):
+            if topic == bayer_image_topic:
+                try:
+                    image = bridge.imgmsg_to_cv2(msg, "mono8")
+                except (CvBridgeError) as e:
+                    print(e)
+                gray_image = cv2.cvtColor(image, cv2.COLOR_BAYER_GR2GRAY)
```

Comment:
I would appreciate a comment here, something like:

"We suspect this hard-coded RGB color mapping is not correct for the Astrobee NavCam/DockCam camera Bayer pattern. It may have swapped color channels. A color channel swap has a subtle effect on grayscale output because different color channels have different weighting factors when outputting luminance calibrated for human perception. However, for the purposes of this script, the key requirement is to exactly replicate the onboard debayer conversion performed in the FSW is_camera ROS node, so localization features will be the same regardless of which tool is used to do the conversion."

Reply:
Done